tests/test_visualizations.py

```diff
@@ -0,0 +1,308 @@
+    content = {
+        "_type": "FilterLayer",
+        "filters": [],
+        "invert": False
+    }
+    return convert_to_analyzere_object(content, LayerView)
+
+
+class TestDescriptionFormatter:
+    def test_description_with_colon(self):
+        raw = 'auto: FilterLayer USHU'
+        expected = 'auto: FilterLayer USHU'
+        assert visualizations._format_description(raw) == expected
+
+    def test_description_with_windows_file_path(self):
+        raw = 'Layer 2 : Z:\\losses\\alpha.csv'
+        expected = 'Layer 2 : Z:\\\\losses\\\\alpha.csv'
+        actual = visualizations._format_description(raw)
+        assert (actual == expected)
+
+    def test_description_with_single_quotes_and_colon(self):
+        raw = ("Layer 'auto: Filter by HU' loaded by 1.25")
+        expected = 'Layer auto: Filter by HU loaded by 1.25'
+        assert visualizations._format_description(raw) == expected
+
+
+class TestDateFieldFormatter:
+    def test_format_DateField(self):
+        expected = '2015-06-01'
+        date = datetime(2015, 6, 1, 12, 11, 10)
+        assert (visualizations._format_DateField(date) == expected)
+
+
+class TestMoneyFieldFormatter:
+    def test_small_MoneyField(self):
+        m = {'currency': 'USD', 'value': 123.11}
+        mf = convert_to_analyzere_object(m)
+        expected = '123 USD'
+        assert visualizations._format_MoneyField(mf) == expected
+
+    def test_large_MoneyField(self):
+        m = {'currency': 'USD', 'value': 123456789.11}
+        mf = convert_to_analyzere_object(m)
+        expected = '123,456,789 USD'
+        assert visualizations._format_MoneyField(mf) == expected
+
+    def test_unlimited_MoneyField(self):
+        m = {'currency': 'USD', 'value': float_info.max}
+        mf = convert_to_analyzere_object(m)
+        expected = 'unlimited'
+        assert visualizations._format_MoneyField(mf) == expected
+
+class TestLayerTermsFormatter:
+    def test_layer(self, layer_view):
+        t,w = visualizations._format_layer_terms(layer_view)
+        assert w == False
+
+    def test_attachment_warning(self, layer_view_attachment_warning):
+        t, w = visualizations._format_layer_terms(
+            layer_view_attachment_warning)
+
+    def test_filter_warning(self, layer_view_filter_warning):
+        t, w = visualizations._format_layer_terms(layer_view_filter_warning)
+        assert w == True
+
+    def test_participation_warning(self, layer_view_participation_warning):
+        t, w = visualizations._format_layer_terms(
+            layer_view_participation_warning)
+        assert w == True
+
+
+class TestLayerViewDigraph:
+    def test_invalid_lv(self):
+        m = {'something': 'here'}
+        lv = convert_to_analyzere_object(m, Resource)
+        with pytest.raises(ValueError):
+            lvg = visualizations.LayerViewDigraph(lv)
+
+    def test_basic(self, layer_view):
+        lvg = visualizations.LayerViewDigraph(layer_view)
```

Comment:
unused local.  could just 
`assert visualizations.LayerViewDigraph(layer_view)` if you expect it to return not None